website/guides/sgb_border.md

```diff
@@ -0,0 +1,178 @@
+# Adding a custom SGB border
+
+This document was written to aid developers of DMG compatibile homebrew with adding Super Game Boy borders.
+
+We wil see how:
+
+- Detect whether we are running on a SGB
+- Transfer the border's tiles
+- Transfer and display the border's tilemap and palettes
+
+<small>
+
+Written by [sylvie (zlago)](https://zlago.github.io/me/), idea (and minor help) by [valentina (coffee bat)](https://coffeebat.neocities.org/), reviews and improvements by [ISSOtm](https://eldred.fr) and [avivace](https://github.com/avivace).
+
+</small>
+
+## Enabling SGB features
+
+Before we can do anything else, we must first specify in the header that this game is an SGB game.
+Otherwise, the SGB BIOS will ignore any packets we send, and we won't even be able to detect when we are running form a SGB.
+
+To enable SGB features:
+
+- The SGB flag (`$0146`) must be set to `$03`;
+- The old licensee code (`$014b`) must be set to `$33`.
+
+This can be achieved by passing the `--sgb-compatible` and `--old-licensee 0x33` flags to [`rgbfix`](https://rgbds.gbdev.io/docs/rgbgfx.1).
+
+## Packets
+
+The SGB BIOS can be "talked to" via *command packets*, sent bit by bit via the `P1`/`JOYP` register.
+
+An SGB packet consists of:
+
+- a "reset pulse" (`P1`=`%xx00xxxx`)
+- 128 data pulses (`P1`=`%xx01xxxx` for "1", `P1`=`%xx10xxxx` for "0")
+- a "0" pulse (`P1`=`%xx10xxxx`)
+
+You must set `P1` to `%xx11xxxx` between each pulse.
+
+This adds up to 16 Bytes of data (LSB first), If a packet doesn't read all 16 bytes, the unused bytes are ignored.
+
+**You should wait 4 frames between each packet.**
+
+For an example of such routine, see this [code](https://github.com/zlago/violence-gbc/blob/11cfdb6ee8a35e042fa9712484d814e0961cea7c/src/sub.sm83#L413-L463) and the related Pan Docs entry: [SGB Command Packet on Pandocs](https://gbdev.io/pandocs/SGB_Command_Packet.html).
+
+This guide glosses over a minor detail, as certain packets can be (albeit unccomon) more than 16 bytes.
```

Comment:
Minor: should additional references/links be put here (e.g. a use case/example of sending more than 16 bytes?) or is it irrelevant/superficial?